### PR TITLE
Refacto/add type util

### PR DESCRIPTION
**Pull request by KevinArtus**

### CODE REVIEW

File: src/Gaufrette/Util/Size.php [hunk 1]
```diff
@@ -41,10 +33,8 @@ public static function fromFile($filename)
      * Returns the size in bytes from the given resource.
      *
      * @param resource $handle
-     *
-     * @return string
      */
-    public static function fromResource($handle)
+    public static function fromResource($handle): int
```
Comment (muchafm) [inline src/Gaufrette/Util/Size.php:37]: I don't think this one is right!

**Review by muchafm — CHANGES_REQUESTED** [review]

File: src/Gaufrette/Util/Size.php (hunk 1 above)
Comment (KevinArtus) [reply src/Gaufrette/Util/Size.php:37]: PHP doc specified int
https://www.php.net/manual/en/function.mb-strlen.php

Comment (muchafm) [reply src/Gaufrette/Util/Size.php:37]: I don't see `mb_strlen` used in this function

File: src/Gaufrette/Util/Checksum.php [hunk 2]
```diff
@@ -5,30 +5,22 @@
...
 
     /**
      * Returns the checksum of the specified file.
-     *
-     * @param string $filename
-     *
-     * @return string
      */
-    public static function fromFile($filename)
+    public static function fromFile(string $filename): false|string
```
Comment (PedroTroller) [inline src/Gaufrette/Util/Checksum.php:23]: `false` return typehint is only available with php 8.2

File: src/Gaufrette/Util/Size.php [hunk 3]
```diff
@@ -27,12 +23,8 @@ public static function fromContent($content)
 
     /**
      * Returns the size in bytes from the given file.
-     *
-     * @param string $filename
-     *
-     * @return int
      */
-    public static function fromFile($filename)
+    public static function fromFile(string $filename): false|int
```
Comment (PedroTroller) [inline src/Gaufrette/Util/Size.php:27]: Same here, `false` return typehint is only available with php 8.2

File: src/Gaufrette/Util/Size.php [hunk 4]
```diff
@@ -41,10 +33,8 @@ public static function fromFile($filename)
      * Returns the size in bytes from the given resource.
      *
      * @param resource $handle
-     *
-     * @return string
      */
-    public static function fromResource($handle)
+    public static function fromResource($handle): array|false|int
```
Comment (PedroTroller) [inline src/Gaufrette/Util/Size.php:37]: Same here

**Review by PedroTroller — CHANGES_REQUESTED** [review]

**Review by muchafm — APPROVED** [review]

File: src/Gaufrette/Util/Size.php [hunk 5]
```diff
@@ -41,10 +33,8 @@ public static function fromFile($filename)
      * Returns the size in bytes from the given resource.
      *
      * @param resource $handle
-     *
-     * @return string
      */
-    public static function fromResource($handle)
+    public static function fromResource($handle): array|bool|int
```
Comment (PedroTroller) [inline src/Gaufrette/Util/Size.php:37]: Why can a size be a bool or a string?

**Review by PedroTroller — CHANGES_REQUESTED** [review]

**Review by PedroTroller — APPROVED** [review]

**Pull request closed without merging**